### ISSUE TITLE
Mulit element POTCAR loading broken.

### ISSUE THREAD

**Issue by MichaelWolloch** [materialsproject/pymatgen#2895]
**Describe the bug**
Since commit e4bde8e400d28095425d0b1e7f0b1c8923ddc13a  it seems that loading is broken if any but the first POTCAR in the file has a sha256 hash.

I really appreciate the effort by @shyuep to clean up the code and get rid of the regex stuff and unicode handling. However, a small mistake appears to have slipped in in line 2277 of `pymatgen/io/vasp/inputs.py`. There should be a unicode end of line character still:
~~~
potcar_strings = fdata.split("End of Dataset\n")
~~~
instead of 
~~~
potcar_strings = fdata.split("End of Dataset")
~~~

If that is included, also the test that got modified in the commit can be changed back it seems.

However, since this all seems a bit fragile, I suggest to actually also downgrade the ValueError that is raised in line 1829 in  `pymatgen/io/vasp/inputs.py` to a warning. 

**To Reproduce**
load the attached POTCAR (containing O and Fe_pv_with_hash from `pymatgen/test_files/POT_GGA_PAW_PBE_54` with the Fe Potcar following the O) into a Potcar object:
[POTCAR_test.gz](https://github.com/materialsproject/pymatgen/files/10982996/POTCAR_test.gz)
~~~
from pymatgen.io.vasp.inputs import Potcar
pot = Potcar.from_file('POTCAR_test.gz')
~~~
results in:
~~~
ValueError: POTCAR with symbol Fe_pv and functional
PBE has a SHA256 hash defined,
but the computed hash differs.
YOUR POTCAR FILE HAS BEEN CORRUPTED AND SHOULD NOT BE USED!
~~~

**Expected behavior**
There should be no error for the SHA256, since the POTCAR is good.

**Environment (please supply relevant versions and platform info):**

- OS: Linux
- Version 2023.3.10

**Additional context**
I am happy to make a pull request if this is wanted.

**Comment by e-kwsm** [comment]
This is not fixed yet (2023.9.25).

```
~/.local/lib/python3.11/site-packages/pymatgen/io/vasp/inputs.py:1806: UnknownPotcarWarning: POTCAR data with symbol O does not match any VASP POTCAR known to pymatgen. There is a possibility your POTCAR is corrupted or that the pymatgen database is incomplete.
  warnings.warn(
~/.local/lib/python3.11/site-packages/pymatgen/io/vasp/inputs.py:1813: UserWarning: POTCAR with symbol Fe_pv and functional
PBE has a SHA256 hash defined,
but the computed hash differs.
YOUR POTCAR FILE HAS BEEN CORRUPTED AND SHOULD NOT BE USED!
  warnings.warn(
```

https://github.com/materialsproject/pymatgen/blob/4bff21d33625bc9f7077c7b4eb06098d1f50277c/pymatgen/io/vasp/inputs.py#L1970

The hashes of POTCAR_test.gz in the OP are `5d22e414b1f82158bf2c7ecb8b97b28fd0923e48cadc1c3bf74d524558f5dd32` and `52459ef0a41f1e32977d35179f8b492c701fda3d7e2f9b778c5c2267c5f2dbc0`.
The former is correct:

```console
$ grep SHA256 Fe_pv/POTCAR
   SHA256 =  5d22e414b1f82158bf2c7ecb8b97b28fd0923e48cadc1c3bf74d524558f5dd32 Fe_pv/POTCAR
$ sed -e /COPYR/d -e /SHA256/d Fe_pv/POTCAR | sha256sum -
5d22e414b1f82158bf2c7ecb8b97b28fd0923e48cadc1c3bf74d524558f5dd32  -
```

The latter is however strangely computed, which can be reproduced by

```console
$ { sed -e /COPYR/d -e /SHA256/d -e '/PAW_PBE/s/^ *//' -e '/TITEL/s/^/   /' -e '/End of/s/^ *//' Fe_pv/POTCAR; printf ''; } | sha256sum -
52459ef0a41f1e32977d35179f8b492c701fda3d7e2f9b778c5c2267c5f2dbc0  -
```

**Comment by MichaelWolloch** [comment]
Hi @e-kwsm !

AFAIK there were some more changes to this since March. There is a pending PR that should fix this along another thing: #3204.

However, POTCAR validation will likely be changed completely anyhow, making this a mute problem. See PR #3351.

**Comment by e-kwsm** [comment]
I am not sure #3351 intends to address this, or checksum will be dropped in future, but anyway SHA256 computation is still (7cc629df501045d53857f26b59348f08a9da3fba) broken (https://github.com/materialsproject/pymatgen/issues/2895#issuecomment-1738275801).
